### PR TITLE
Add GSSAPI support to Koji integration

### PR DESCRIPTION
**Pull request by ondrejbudai**
Working GSSAPI support to Koji integration. With tests, refactoring and all other nice things.

### CODE REVIEW

File: internal/upload/koji/run-koji-container.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ ${CONTAINER_RUNTIME} run -d --name org.osbuild.koji.koji --network org.osbuild.k
   -e POSTGRES_PASSWORD=kojipass \
   -e POSTGRES_DB=koji \
   -e POSTGRES_HOST=org.osbuild.koji.postgres \
-  quay.io/osbuild/ghci-koji:rc1
+  quay.io/osbuild/ghci-koji:v1
```
Comment (teg) [inline internal/upload/koji/run-koji-container.sh]: Intentional, or did we do something wrong? @dvdhrm?

Comment (teg) [reply internal/upload/koji/run-koji-container.sh]: That `rc1` doesn't exist any more, I mean.

Comment (ondrejbudai) [reply internal/upload/koji/run-koji-container.sh]: No idea. The primary issue is that I left there `rc1` for many months, I should have changed it to `v1` but forgot.

File: internal/upload/koji/koji.go [hunk 2]
```diff
@@ -106,46 +106,53 @@ type CGImportResult struct {
 	BuildID int `xmlrpc:"build_id"`
 }
 
-func New(server string) (*Koji, error) {
-	k := &Koji{}
-	client, err := xmlrpc.NewClient(server, http.DefaultTransport)
+func Login(server, user, password string) (*Koji, error) {
```
Comment (teg) [inline internal/upload/koji/koji.go]: Nitpick for future work: Should this be called `New()`?

File: internal/upload/koji/koji.go [hunk 3]
```diff
@@ -117,34 +151,36 @@ func Login(server, user, password string, transport http.RoundTripper) (*Koji, e
...
-	// sessionID, sessionKey and callnum
-	kojiTransport := &Transport{
-		sessionID:  reply.SessionID,
-		sessionKey: reply.SessionKey,
-		callnum:    0,
-		transport:  transport,
+	return newKoji(server, transport, reply)
+}
+
+func GSSAPILogin(server string, credentials *GSSAPICredentials, transport http.RoundTripper) (*Koji, error) {
```
Comment (teg) [inline internal/upload/koji/koji.go:163]: Further to my previous comment, maybe name the constructors `NewFromPlain()` and `NewFromGSSAPI()`, or whatever the canonical names of these things are ;)

File: internal/upload/koji/koji_test.go [hunk 4]
```diff
@@ -22,17 +24,36 @@ import (
 
 func TestKojiImport(t *testing.T) {
 	// define constants
-	server := "http://localhost:8080/kojihub"
-	user := "osbuild"
-	password := "osbuildpass"
+	server := "https://localhost/kojihub"
 	filename := "image.qcow2"
 	filesize := 1024
+	shareDir := "/tmp/osbuild-composer-koji-test"
```
Comment (teg) [inline internal/upload/koji/koji_test.go:30]: Use a `TmpDir()` to avoid clashes with other test runs?

File: internal/upload/koji/koji.go (hunk 2 above)
Comment (ondrejbudai) [reply internal/upload/koji/koji.go]: Fixed

File: internal/upload/koji/koji.go (hunk 3 above)
Comment (ondrejbudai) [reply internal/upload/koji/koji.go]: Fixed

File: internal/upload/koji/koji_test.go (hunk 4 above)
Comment (ondrejbudai) [reply internal/upload/koji/koji_test.go:30]: I think I don't care for now, this tempdir is shared between the test and the container runner. Sharing the tempdir path would be more pain than I currently want.

**Comment by ondrejbudai** [comment]
This is now ready to be reviewed and merged! :partying_face:

**Comment by teg** [comment]
Integration tests on RHEL8 unrelated, this only touches unit tests.

**Review by teg — APPROVED** [review]
Love the cleanups, love the feature, love the tests!

**Pull request merged**